overlord/snapstate/autorefresh.go

```diff
@@ -149,6 +150,44 @@ func (m *autoRefresh) AtSeed() error {
 	return nil
 }
 
+func (m *autoRefresh) canRefreshRespectingMetered(now, lastRefresh time.Time) (can bool, err error) {
+	can, err = canRefreshOnMeteredConnection(m.state)
+	if err != nil {
+		return false, err
+	}
+	if can {
+		return true, nil
+	}
+
+	// ignore any errors that occurred while checking if we are on a metered
+	// connection
+	metered, _ := IsOnMeteredConnection()
+	if !metered {
+		return true, nil
+	}
+
+	if now.Sub(lastRefresh) >= maxPostponement {
+		// TODO use warnings when the infra becomes available
+		logger.Noticef("Last refresh more than %d days ago, refreshing anyway", int(maxPostponement.Hours()/24))
```

Comment:
Message is a bit curious here. It doesn't say what the "anyway" means. Perhaps something similar to:

_"Auto refresh disabled on metered, but pending for too long (%s days). Trying to refresh now."_